ScanCommand/src/main/java/io/ballerina/scan/ScanCmd.java

```diff
@@ -114,8 +122,43 @@ public void execute() {
             return;
         }
 
+        Project project = getProject();
+        if (project == null) {
+            return;
+        }
+
+        ProjectAnalyzer projectAnalyzer = new ProjectAnalyzer();
+
         outputStream.println();
         outputStream.println("Running Scans...");
+
+        List<Issue> issues = projectAnalyzer.analyzeProject(project);
+
+        if (!platforms.isEmpty() || platformTriggered) {
```

Comment:
Why are we doing this?

Reply:
When platforms to report analysis issues is provided via the `--platforms` flag, or if the scan command is triggered from an external platform, signified by the `--platform-triggered` flag, to directly pass the reported issues to the relevant platform plugin without logging the analysis issues to the console and generating reports in the working directory.